doc/tag_parser.py

```diff
@@ -0,0 +1,143 @@
+from __future__ import print_function
+import sys
+import os
+from collections import defaultdict
+import fnmatch
+
+# mask for files to parse
+file_masks = {'*.py', '*.focs.txt', '*.macros'}
+
+# mask for files to exclude from parse, checked after file_masks
+exclude_masks = {'__*'}
+
+# prefix to initiate a match condition
+tag_comment = "@content_tag{"
+
+# Root to search for files
+source_dir = sys.argv[1]
+
+# Source and target paths for file generation
+doc_dir = os.path.join(source_dir, 'doc')
+dox_source = os.path.join(doc_dir, 'all_tags.dox.in')
+dox_target = os.path.join(doc_dir, 'all_tags.dox')
+
+# match condition in doc_target to replace with result of parsing
+dox_variable = "${CONTENT_DOCUMENTATION}"
+
+# prefix to source links (being the cpp api, internal links do not exist)
+base_link_url = 'https://github.com/freeorion/freeorion/blob/master/'
```

Comment:
The links for python/focs files would not exist on the `.io` site?  Will look into alternative.